service/raft/raft_rpc.cc

```diff
@@ -154,11 +154,12 @@ raft_rpc::on_configuration_change(raft::server_address_set add, raft::server_add
         // Entries explicitly managed via `rpc::add_server` and
         // `rpc::remove_server` should never expire while entries
         // learnt upon receiving an rpc message should be expirable.
-        _address_map.set(addr.id, inet_addr, false);
+        _address_map.add_or_update_entry(addr.id, inet_addr);
```

Comment:
I think you got confused with the patch order. Here we still don't use gossiper and propagate the entries via raft configuration. When we switch to using gossiper this place will change to call set_nonexpiring, since it won't have an ip_addr to share.

Reply:
```
void                                                                                                
raft_rpc::on_configuration_change(raft::server_address_set add, raft::server_address_set del) {     
    for (const auto& addr: add) {                                                                   
        _address_map.set_nonexpiring(addr.id);                                                      
    }                                                                                               
    for (const auto& addr: del) {                                                                   
        _address_map.set_expiring(addr.id);                                                         
    }                                                                              ``                 
} 
```